error.go

```diff
@@ -23,5 +30,9 @@ func normalizeAppErr(appErr *model.AppError) error {
 		return nil
 	}
 
+	if appErr.StatusCode == http.StatusNotFound {
+		return ErrNotFound
+	}
```

Comment:
In the discussion about error handling we decided to not return an error for `List` methods that return an empty list. https://github.com/mattermost/mattermost-plugin-api/issues/22 encapsulate that decisions.

Are we sticking with the decision? If so, can you confirm that no `List` method returns an error?

Reply:
Yes, I'm inclined to stick with that decision -- an empty list is not an error. Do we need to comb through anything in advance of formally addressing https://github.com/mattermost/mattermost-plugin-api/issues/22? If any `List` methods return errors today instead of empty results, this doesn't make that /worse/, strictly speaking.

Reply:
Good point, this PR doesn't make it worse. With that in mind I'm fine with merging.

Yes, we have to comb trough every List method for #22.